### PR TITLE
[CARBONDATA-986] Add alter table example

### PR DESCRIPTION
**Pull request by chenerlu**
Description: Just add alter table example for testing alter table grammar

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1797/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1803/

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/AlterTableExample.scala [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+import java.io.File
+
+import org.apache.carbondata.core.constants.CarbonCommonConstants
+import org.apache.carbondata.core.util.CarbonProperties
+import org.apache.spark.sql.CarbonSession._
+import org.apache.spark.sql.SparkSession
+
+/**
+  * Created by erlu on 2017/4/25.
```
Comment (chenliang613) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/AlterTableExample.scala]: the comment format is incorrect.

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/AlterTableExample.scala [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+         |    floatField float,
+         |    complexData array<string>
+         | )
+         | STORED BY 'carbondata'
+         | TBLPROPERTIES('DICTIONARY_INCLUDE'='dateField, charField')
+       """.stripMargin)
+
+    // Alter table change data type
+    spark.sql("DESCRIBE FORMATTED carbon_table").show()
+    spark.sql("ALTER TABLE carbon_table CHANGE intField intField bigint").show()
```
Comment (chenliang613) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/AlterTableExample.scala]: why here put two "intField"?

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/AlterTableExample.scala [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    spark.sql("DESCRIBE FORMATTED carbon_table").show()
+
+    // Alter table add columns
+    spark.sql("DESCRIBE FORMATTED carbon_table").show()
+    spark.sql("ALTER TABLE carbon_table ADD COLUMNS (newField String)" +
+      " TBLPROPERTIES ('DEFAULT.VALUE.newField'='def')").show()
+    spark.sql("DESCRIBE FORMATTED carbon_table").show()
+
+    // Alter table drop columns
+    spark.sql("DESCRIBE FORMATTED carbon_table").show()
```
Comment (chenliang613) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/AlterTableExample.scala]: this row is not required. same as row83

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/AlterTableExample.scala [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    spark.sql("ALTER TABLE carbon_table DROP COLUMNS (newField)").show()
```
Comment (chenliang613) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/AlterTableExample.scala]: can you change the drop column to other column, not just added column.

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/AlterTableExample.scala (hunk 2 above)
Comment (chenerlu) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/AlterTableExample.scala]: This is Alter table change data type grammar

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1809/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1810/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1812/

**Comment by chenliang613** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1813/

**Pull request closed without merging**